sihl_core/lib/db.ml

```diff
@@ -77,7 +66,9 @@ let key : db_connection Opium.Hmap.key =
 
 let request_with_connection request =
```

Comment:
I think this should go away, but we can change it later. We should design API that allows queries only when transaction is started (explicitly). We've seen what happens when you don't think about this and expect implicit transactions.

Reply:
I think the default should be no transactions and carefully place transactions where needed. Rails, Django and Laravel all are perfectly usable without thinking about transactions. 

What do you propose? Transactions scope is request/response?